PublicSDKAPITester/RevenueCatAPI.m

```diff
@@ -0,0 +1,120 @@
+//
+//  RevenueCatAPI.m
+//  MigrateTester
+//
+//  Created by Joshua Liebowitz on 6/18/21.
+//
+
+#import "RevenueCatAPI.h"
+#import "Purchases.h"
+#import "RCPurchases.h"
+
+@import StoreKit;
+
+@implementation RevenueCatAPI
+
+bool canI;
+NSString *version;
+
+BOOL automaticAppleSearchAdsAttributionCollection;
+BOOL debugLogsEnabled;
+RCLogLevel logLevel;
+NSURL *proxyURL;
+BOOL forceUniversalAppStore;
+BOOL simulatesAskToBuyInSandbox;
+RCPurchases *sharedPurchases;
+BOOL isConfigured;
+BOOL allowSharingAppStoreAccount;
+BOOL finishTransactions;
+id<RCPurchasesDelegate> delegate;
+NSString *appUserID;
+BOOL isAnonymous;
+
++ (void)allTheThings {
+//    typedef void (^RCReceivePurchaserInfoBlock)(RCPurchaserInfo * _Nullable, NSError * _Nullable) NS_SWIFT_NAME(Purchases.ReceivePurchaserInfoBlock);
+//    typedef void (^RCReceiveIntroEligibilityBlock)(NSDictionary<NSString *, RCIntroEligibility *> *) NS_SWIFT_NAME(Purchases.ReceiveIntroEligibilityBlock);
+//    typedef void (^RCReceiveOfferingsBlock)(RCOfferings * _Nullable, NSError * _Nullable) NS_SWIFT_NAME(Purchases.ReceiveOfferingsBlock);
+//    typedef void (^RCReceiveProductsBlock)(NSArray<SKProduct *> *) NS_SWIFT_NAME(Purchases.ReceiveProductsBlock);
+//    typedef void (^RCPurchaseCompletedBlock)(SKPaymentTransaction * _Nullable, RCPurchaserInfo * _Nullable, NSError * _Nullable, BOOL userCancelled) NS_SWIFT_NAME(Purchases.PurchaseCompletedBlock);
+//    typedef void (^RCDeferredPromotionalPurchaseBlock)(RCPurchaseCompletedBlock);
+//    typedef void (^RCPaymentDiscountBlock)(SKPaymentDiscount * _Nullable, NSError * _Nullable) NS_SWIFT_NAME(Purchases.PaymentDiscountBlock);
+
+    //    [p presentCodeRedemptionSheet]; ///TODO: iOS ONLY, TEST FOR THIS API BY LOOKING UP SELECTOR
+    RCPurchases *p = [RCPurchases configureWithAPIKey:@""];
+    
+    [RCPurchases setLogHandler:^(RCLogLevel l, NSString *i) {}];
+    canI = [RCPurchases canMakePayments];
+    version = [RCPurchases frameworkVersion];
+    [RCPurchases addAttributionData:@{} fromNetwork:RCAttributionNetworkBranch];
+    [RCPurchases addAttributionData:@{} fromNetwork:RCAttributionNetworkBranch forNetworkUserId:@""];
+        
+    automaticAppleSearchAdsAttributionCollection = [RCPurchases automaticAppleSearchAdsAttributionCollection];
+    debugLogsEnabled = [RCPurchases debugLogsEnabled];
+    logLevel = [RCPurchases logLevel];
+    proxyURL = [RCPurchases proxyURL];
+    forceUniversalAppStore = [RCPurchases forceUniversalAppStore];
+    simulatesAskToBuyInSandbox = [RCPurchases simulatesAskToBuyInSandbox];
+    sharedPurchases = [RCPurchases sharedPurchases];
+    isConfigured = [RCPurchases isConfigured];
+    allowSharingAppStoreAccount = [p allowSharingAppStoreAccount];
+    finishTransactions = [p finishTransactions];
+    delegate = [p delegate];
+    appUserID = [p appUserID];
+    isAnonymous = [p isAnonymous];
```

Comment:
out of curiosity, why are these defined at the class level as opposed to the method? Given that the rest are in the method

Reply:
No reason, my methodology is:
- Proof of concept
- Integrate work
- Cleanup
- optimize

I just finished `Integrate work` 😄